modules/core/src/main/scala/org/scalasteward/core/edit/UpdateHeuristic.scala

```diff
@@ -91,7 +91,7 @@ object UpdateHeuristic {
       searchTermsToAlternation(getSearchTerms(update).map(removeCommonSuffix)).map { searchTerms =>
         val prefix = getPrefixRegex(update).getOrElse("")
         val currentVersion = Regex.quote(update.currentVersion)
-        s"(?i)(.*)($prefix$searchTerms.*?)$currentVersion(.?)".r
+        s"(?is)(.*?)($prefix$searchTerms(?:(?!\\d+\\.\\d+).)*?)$currentVersion(.?)".r
```

Comment:
Note) `(?:(?!\\d+\\.\\d+).)*?` is a key change.
The regex will try to find a multiline text containing `$currentVersion` and not containing other partial versions in the middle of text.